freqtrade/rpc/rpc.py

```diff
@@ -1115,19 +1115,33 @@ def _rpc_delete(self, trade_id: int) -> dict[str, str | int]:
                 "cancel_order_count": c_count,
             }
 
-    def _rpc_list_custom_data(self, trade_id: int, key: str | None) -> list[dict[str, Any]]:
-        # Query for trade
-        trade = Trade.get_trades(trade_filter=[Trade.id == trade_id]).first()
-        if trade is None:
+    def _rpc_list_custom_data(
+        self, trade_id: int | None = None, key: str | None = None
+    ) -> list[dict[str, Any]]:
+        """
+        Fetch custom data for a specific trade, or all open trades if `trade_id` is not provided.
+        """
+        trades: Sequence[Trade]
+        if trade_id is None:
+            # get all open trades
+            trades = Trade.get_open_trades()
+        else:
+            trades = Trade.get_trades(trade_filter=[Trade.id == trade_id]).all()
+
+        if not trades:
             return []
-        # Query custom_data
+
         custom_data = []
-        if key:
-            data = trade.get_custom_data(key=key)
-            if data:
-                custom_data = [data]
-        else:
-            custom_data = trade.get_all_custom_data()
+        for trade in trades:
+            # Collect custom data
+            if key:
+                data = trade.get_custom_data(key=key)
+                if data:
+                    custom_data = [data]
```

Comment:
i don't see any indication that passing "key" restricts this method to a single trade.
with the current implementation however - this becomes an (implicit) fact - so if i call `_rpc_list_custom_data(key="hello")` - i get the data for the very last trade (based on no sorting) - while in reality, i asked for info for all trades (if it exists, obviously).

The fact that tests pass right now highlights the fact that this is also missing in test coverage.